src/sync.rs

```diff
@@ -713,6 +729,47 @@ fn test_non_lockfree_unchecked() {
     LockFreeFrozenVec::<()>::new();
 }
 
+impl<T: Copy + Clone> Clone for LockFreeFrozenVec<T> {
+    fn clone(&self) -> Self {
+        let len = self.len.load(Ordering::Acquire);
+        // handle the empty case
+        if len == 0 {
+            return Self::default();
+        }
```

Comment:
Not sure this code path is worth it. If we leave this code out, the logic below will do one atomic (acquire) load of a pointer and check it against null. So maybe just remove it

Reply:
Fixed in [`50d0bf8` (#57)](https://github.com/Manishearth/elsa/pull/57/commits/50d0bf873aa8a6b18f38d6e868b3b400ac733380)